frontends/web/jest.config.js

```diff
@@ -0,0 +1,227 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// For a detailed explanation regarding each configuration property, visit:
+// https://jestjs.io/docs/en/configuration.html
+
+module.exports = {
+    // All imported modules in your tests should be mocked automatically
+    // automock: false,
+
+    // Stop running tests after the first failure
+    // bail: false,
+
+    // Respect "browser" field in package.json when resolving modules
+    // browser: false,
+
+    // The directory where Jest should store its cached dependency information
+    // cacheDirectory: "/tmp/jest_rs",
+
+    // Automatically clear mock calls and instances between every test
+    clearMocks: true,
+
+    // Indicates whether the coverage information should be collected while executing the test
+    // collectCoverage: false,
+
+    // An array of glob patterns indicating a set of files for which coverage information
+    // should be collected.
+    collectCoverageFrom: [
+        'src/**/*.{js,jsx,ts,tsx}'
+    ],
+
+    // The directory where Jest should output its coverage files
+    coverageDirectory: 'coverage',
+
+    // An array of regexp pattern strings used to skip coverage collection
+    // coveragePathIgnorePatterns: [
+    //   "/node_modules/"
+    // ],
+
+    // A list of reporter names that Jest uses when writing coverage reports
+    // coverageReporters: [
+    //   "json",
+    //   "text",
+    //   "lcov",
+    //   "clover"
+    // ],
+
+    // An object that configures minimum threshold enforcement for coverage results
+    // coverageThreshold: null,
+
+    // Make calling deprecated APIs throw helpful error messages
+    // errorOnDeprecated: false,
+
+    // Force coverage collection from ignored files usin a array of glob patterns
+    // forceCoverageMatch: [],
+
+    // A path to a module which exports an async function that is triggered once
+    // before all test suites.
+    // globalSetup: null,
+
+    // A path to a module which exports an async function that is triggered once
+    // after all test suites.
+    // globalTeardown: null,
+
+    // A set of global variables that need to be available in all test environments
+    globals: {
+        'ts-jest': {
+            // Need to enable --esModuleInterop so that JSX components can look up CSS styles.
+            // Additionally, modules import/export needs to be set back commonjs.
+            // Both requirements are due to the tests being run via ts-jest.
+            tsConfigFile: 'tsconfig.test.json'
+        }
+    },
+
+    // An array of directory names to be searched recursively up from the requiring module's location
+    // moduleDirectories: [
+    //   "node_modules"
+    // ],
+
+    // An array of file extensions the project modules use.
+    moduleFileExtensions: [
+        'js',
+        'json',
```

Comment:
My understanding of what we do here is quite limited but why do we include `json`? And shouldn't you add `ts`?

Reply:
`json` is for making i18n's translations available as modules during tests.
re: `.ts`. I don't believe we have any .ts files, so didn't add it. is the plan to also switch to plain typescript? if so, need to add it here. otherwise, I don't see why.

Reply:
I have to admin though, this is a bit inconsistent with the extensions listed in `collectCoverageFrom` above.

Reply:
Actually, you're right. We do use .ts files: the generated xxx.d.ts from CSS files.
Will add `ts` to this list.

Reply:
done